dev/benchmarks/microbenchmarks/lib/foundation/observer_list_bench.dart

```diff
@@ -0,0 +1,181 @@
+
+      final ObserverList<VoidCallback> observerList =
+          ObserverList<VoidCallback>();
+      for (int i = 0; i < callbackCount; ++i) {
+        observerList.add(
+          switch (failRemoval) {
+            false => () {
+                final VoidCallback first =
+                    (observerList.iterator..moveNext()).current;
+
+                observerList.remove(first);
+                observerList.add(first);
+              },
+            true => () => observerList.remove(miss),
+          },
+        );
+      }
+
+      final Stopwatch watch = Stopwatch()..start();
+
+      for (int i = 0; i < iterations; ++i) {
+        final List<VoidCallback> list = observerList.toList(growable: false);
+        for (final VoidCallback cb in list) {
+          if (observerList.contains(cb)) {
+            cb();
+          }
+        }
+      }
+
+      watch.stop();
+
+      if (addResult) {
+        printer.addResult(
+          description: '$name ($callbackCount callbacks)',
+          value: watch.elapsedMicroseconds / iterations,
+          unit: 'µs per iteration',
+          name: '$name$callbackCount',
+        );
+      }
+    }
+  }
+
+  void runNotifiyListenersLoopWithHashedObserverList(
+    int totalIterations, {
+    bool addResult = true,
+  }) {
+    const String name = 'notifyListeners:HashedObserverList';
+
+    for (final int callbackCount in callbackCounts) {
+      final int iterations = totalIterations ~/ callbackCount;
+
+      final HashedObserverList<VoidCallback> observerList =
+          HashedObserverList<VoidCallback>();
+      for (int i = 0; i < callbackCount; ++i) {
+        observerList.add(() {
+          final VoidCallback first =
+              (observerList.iterator..moveNext()).current;
+
+          observerList.remove(first);
+          observerList.add(first);
+        });
+      }
+
+      final Stopwatch watch = Stopwatch()..start();
+
+      for (int i = 0; i < iterations; ++i) {
+        final List<VoidCallback> list = observerList.toList(growable: false);
+        for (final VoidCallback cb in list) {
+          if (observerList.contains(cb)) {
+            cb();
+          }
+        }
+      }
+
+      watch.stop();
+
+      if (addResult) {
+        printer.addResult(
+          description: '$name ($callbackCount callbacks)',
+          value: watch.elapsedMicroseconds / iterations,
+          unit: 'µs per iteration',
+          name: '$name$callbackCount',
+        );
+      }
+    }
+  }
+
+  void runNotifiyListenersLoopWithAnimationController(
+    int totalIterations, {
+    bool addResult = true,
+  }) {
+    const String name = 'notifyListeners:AnimationController';
+
+    for (final int callbackCount in callbackCounts) {
+      final int iterations = totalIterations ~/ callbackCount;
+
+      final TestAnimationController controller = TestAnimationController();
+      for (int i = 0; i < callbackCount; ++i) {
+        late final VoidCallback cb;
```

Comment:
This doesn't have to be late?

Reply:
Rather `late` than `Never`, is what the compiler would say.